src/facet.js

```diff
@@ -38,24 +38,29 @@ class Facet extends Mark {
     }
     for (let i = 0; i < this.marks.length; ++i) {
       const mark = this.marks[i];
-      const markFacets = mark.data === this.data ? facetsIndex : undefined;
-      const {index, channels} = mark.initialize(markFacets);
+      const markFacets = mark.facet == null ? mark.data === this.data ? facetsIndex : undefined
+        : mark.facet === "exclude" ? facetsIndex.map(facet => Uint32Array.from(difference(index, facet)))
+        : mark.facet === true ? facetsIndex
+        : undefined;
+      if (mark.facet && range(mark.data).length !== index.length) {
```

Comment:
We should use the same test we use below:

```suggestion
      if (markFacets && range(mark.data).length !== index.length) {
```